Documentation/format/index.md

```diff
@@ -0,0 +1,198 @@
+│ │                  │ ┌──────────────────────────┐     │ │
+│ │                  │ │ ref(l_i.name) <varint>   │     │ │
+│ │ #labels <varint> │ ├──────────────────────────┤ ... │ │
+│ │                  │ │ ref(l_i.value) <varint>  │     │ │
+│ │                  │ └──────────────────────────┘     │ │
+│ ├──────────────────┼──────────────────────────────────┤ │
+│ │                  │ ┌──────────────────────────┐     │ │
+│ │                  │ │ c_i.mint <varint>        │     │ │
+│ │                  │ ├──────────────────────────┤     │ │
+│ │                  │ │ c_i.maxt <varint>        │     │ │
+│ │ #chunks <varint> │ ├──────────────────────────┤ ... │ │
+│ │                  │ │ ref(c_i.data) <varint>   │     │ │
+│ │                  │ ├──────────────────────────┤     │ │
+│ │                  │ │ crc32(c_i.data) <varint> │     │ │
+│ │                  │ └──────────────────────────┘     │ │
+│ └──────────────────┴──────────────────────────────────┘ │
+├─────────────────────────────────────────────────────────┤
+│ CRC32 <4 byte>                                          │
+└─────────────────────────────────────────────────────────┘
+```
+
+The CRC checksum is calculated over the series contents of the index concatenated with the data of its chunks (with encoding byte, without length).
+
+
+### Label Index
+
+The label index indexes holds lists of possible values for label names. A sequence of label index blocks follow on the series entries.
+
+```
+┌─────────────────────────────────────────────────────────┐
+│ len <varint>                                            │
+├─────────────────────────────────────────────────────────┤
+│ ┌──────────────────┬──────────────────────────────────┐ │
+│ │                  │ ┌──────────────────────────┐     │ │
+│ │                  │ │ ref(value[0]) <4 byte>   │     │ │
+│ │                  │ ├──────────────────────────┤     │ │
+│ │  n = len(names)  │ │ ...                      │ ... │ │
+│ │     <varint>     │ ├──────────────────────────┤     │ │
+│ │                  │ │ ref(value[n]) <4 byte>   │     │ │
+│ │                  │ └──────────────────────────┘     │ │
+│ └──────────────────┴──────────────────────────────────┘ │
+├─────────────────────────────────────────────────────────┤
+│ CRC32 <4 byte>                                          │
+└─────────────────────────────────────────────────────────┘
+```
+
+The sequence of label index blocks is finalized by a lookup table pointing to the beginning of each label index block. It is simply a list of entries that are read into an in-memory hashmap when the index is loaded.
+
+### Postings
+
+Postings are postings lists that map label pairs to series they occur in.
+
+```
+┌─────────────────────────────────────────────────┐
+│ len <varint>                                    │
+├─────────────────────────────────────────────────┤
+│ ┌─────────────────────────────────────────────┐ │
+│ │ ref(series[0]) <4 byte>                     │ │
```

Comment:
Because our `Seek()` method does binary search, which needs constant spacing.
If we go with compressed postings, e.g. by using delta varint encoding, we need a skip table unless we want to always decode every single entry.

Suppose you want to intersect two postings and one start with 1,000,000 and the other one with 1. We potentially have to decode 1e6 entries before even starting with intersecting properly.